src/ray/util/event.cc

```diff
@@ -253,28 +270,39 @@ void RayEvent::SendMessage(const std::string &message) {
                                        ? RayEventContext::Instance()
                                        : RayEventContext::GlobalInstance();
 
-  rpc::Event event;
-
-  std::string event_id_buffer = std::string(18, ' ');
-  FillRandom(&event_id_buffer);
-  event.set_event_id(StringToHex(event_id_buffer));
-
-  event.set_source_type(context.GetSourceType());
-  event.set_source_hostname(context.GetSourceHostname());
-  event.set_source_pid(context.GetSourcePid());
-
-  event.set_severity(severity_);
-  event.set_label(label_);
-  event.set_message(message);
-  event.set_timestamp(current_sys_time_us());
+  static const int event_id_size = 18;
+  static const std::string empty_event_id_hex(event_id_size * 2, 'f');
+  std::string event_id;
+  if (IsLevelEnabled(severity_)) {
+    std::string event_id_buffer = std::string(event_id_size, ' ');
+    FillRandom(&event_id_buffer);
+    event_id = StringToHex(event_id_buffer);
+    rpc::Event event;
+    event.set_event_id(event_id);
+
+    event.set_source_type(context.GetSourceType());
+    event.set_source_hostname(context.GetSourceHostname());
+    event.set_source_pid(context.GetSourcePid());
+
+    event.set_severity(severity_);
+    event.set_label(label_);
+    event.set_message(message);
+    event.set_timestamp(current_sys_time_us());
+
+    auto mp = context.GetCustomFields();
+    for (const auto &pair : mp) {
+      custom_fields_[pair.first] = pair.second;
+    }
+    event.mutable_custom_fields()->insert(mp.begin(), mp.end());
 
-  auto mp = context.GetCustomFields();
-  for (const auto &pair : mp) {
-    custom_fields_[pair.first] = pair.second;
+    EventManager::Instance().Publish(event, custom_fields_);
+  } else {
+    event_id = empty_event_id_hex;
+  }
+  if (ray::RayLog::IsLevelEnabled(log_severity_)) {
+    ::ray::RayLog(__FILE__, __LINE__, log_severity_)
```

Comment:
will __FILE__, __LINE__ always be event.cc:303? This is a little bit inconvenient.